docs/spec/distributing.rst

```diff
@@ -266,3 +622,23 @@ of that Python version. This can be queried e.g.
 ``pythonX.Y -c 'import site; print(site.getsitepackages())'``. It is also recommended
 that the type checker allow for the user to point to a particular Python
 binary, in case it is not in the path.
+
+.. _import-conventions:
+
+Import Conventions
+------------------
+
+By convention, certain import forms indicate to type checkers that an imported
+symbol is re-exported and should be considered part of the importing module's
+public interface. All other imported symbols are considered private by default.
+
+The following import forms re-export symbols:
+
+* ``import X as X`` (a redundant module alias): re-exports ``X``.
+* ``from Y import X as X`` (a redundant symbol alias): re-exports ``X``.
+* ``from Y import *``: re-exports all symbols in ``Y`` that do not begin with
```

Comment:
I think this is wrong. All symbols imported by a wildcard (as indicated by the `__all__` symbol) should be re-exported, including those that begin with an underscore. See [this documentation](https://github.com/python/typing/blob/main/docs/guides/libraries.rst#library-interface-public-and-private-symbols) for details.

Reply:
If there is no `__all__`, then `import *` does re-export all names that do not start with an underscore (by analogy with Python's runtime behavior).

Reply:
Yes, I agree that's the correct behavior if there's no `__all__`. That matches pyright's behavior. Not sure about other type checkers.

Reply:
Updated this bullet point to describe the behavior when `__all__` is present.